modules/js-and-css/speculation-rules/settings.php

```diff
@@ -0,0 +1,218 @@
+ * Sanitizes the setting for Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ *
+ * @param mixed $input Setting to sanitize.
+ * @return array Sanitized setting, an associative array with 'mode' and 'eagerness' keys.
+ */
+function plsr_sanitize_setting( $input ) {
+	$default_value = plsr_get_setting_default();
+
+	if ( ! is_array( $input ) ) {
+		return $default_value;
+	}
+
+	$mode_labels      = plsr_get_mode_labels();
+	$eagerness_labels = plsr_get_eagerness_labels();
+
+	// Ensure only valid keys are present.
+	$value = array_intersect_key( $input, $default_value );
+
+	// Set any missing or invalid values to their defaults.
+	if ( ! isset( $value['mode'] ) || ! isset( $mode_labels[ $value['mode'] ] ) ) {
+		$value['mode'] = $default_value['mode'];
+	}
+	if ( ! isset( $value['eagerness'] ) || ! isset( $eagerness_labels[ $value['eagerness'] ] ) ) {
+		$value['eagerness'] = $default_value['eagerness'];
+	}
+
+	return $value;
+}
+
+/**
+ * Registers setting to control Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+function plsr_register_setting() {
+	register_setting(
+		'reading',
+		'plsr_speculation_rules',
+		array(
+			'type'              => 'object',
+			'description'       => __( 'Configuration for the Speculation Rules API.', 'performance-lab' ),
+			'sanitize_callback' => 'plsr_sanitize_setting',
+			'default'           => plsr_get_setting_default(),
+			'show_in_rest'      => array(
+				'schema' => array(
+					'properties' => array(
+						'mode'      => array(
+							'description' => __( 'Whether to prefetch or prerender URLs.', 'performance-lab' ),
+							'type'        => 'string',
+							'enum'        => array_keys( plsr_get_mode_labels() ),
+						),
+						'eagerness' => array(
+							'description' => __( 'Whether to trigger on click (conservative), on hover (moderate), or on even the slight suggestion (eager) that the user may navigate to the URL.', 'performance-lab' ),
+							'type'        => 'string',
+							'enum'        => array_keys( plsr_get_eagerness_labels() ),
+						),
+					),
+				),
+			),
+		)
+	);
+}
+add_action( 'init', 'plsr_register_setting' );
+
+/**
+ * Adds the settings sections and fields for the Speculation Rules configuration.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+function plsr_add_setting_ui() {
+	add_settings_section(
+		'plsr_speculation_rules',
+		__( 'Speculation Rules', 'performance-lab' ),
+		static function () {
+			?>
+			<p class="description">
+				<?php esc_html_e( 'This section allows you to control how URLs that your users navigate to are speculatively loaded to improve performance.', 'performance-lab' ); ?>
+			</p>
+			<?php
+		},
+		'reading'
+	);
+
+	$fields = array(
+		'mode'      => array(
+			'title'       => __( 'Mode', 'performance-lab' ),
+			'description' => __( 'Prerendering will lead to faster load times than prefetching, however in case of interactive content prefetching may be a safer choice.', 'performance-lab' ),
```

Comment:
grammar nit

```suggestion
			'description' => __( 'Prerendering will lead to faster load times than prefetching. However, in case of interactive content, prefetching may be a safer choice.', 'performance-lab' ),
```

There are other aspects to the tradeoff (resource utilization, so you can prefetch more links than you can prerender, etc). But we can only fit so much nuance here, so this might suffice.